src/Tickets/Commerce/Payments_Tab.php

```diff
@@ -195,117 +216,136 @@ public function get_section_menu() {
 	 * @return string
 	 */
 	public function filter_redirect_url( $url ) {
+		if ( ! is_admin() ) {
+			return $url;
+		}
+
+		$tab  = tribe_get_request_var( 'tab' );
+		$page = tribe_get_request_var( 'page' );
+
+		if ( empty( $tab ) || empty( $page ) ) {
+			return $url;
+		}
 
-		// Parse URL to get query string info.
-		$url_query = wp_parse_url( $url, PHP_URL_QUERY );
-		wp_parse_str( $url_query, $args );
+		if ( Tribe__Settings::$parent_slug !== $page ) {
+			return $url;
+		}
 
-		// If not on the TEC Payments tab, bail.
-		if (
-			( empty( $args['page'] ) || Tribe__Settings::$parent_slug !== $args['page'] ) ||
-			( empty( $args['tab'] )  || static::$slug !== $args['tab'] )
-		) {
+		if ( static::$slug !== $tab ) {
 			return $url;
 		}
 
-		// If valid section not posted, bail.
-		$current_section_key = Payments_Tab::$key_current_section;
-		if ( empty( $_POST[ $current_section_key ] ) || 'main' === $_POST[ $current_section_key ] ) {
+		$section = tribe_get_request_var( static::$key_current_section );
+		if ( empty( $section ) ) {
+			$section = tribe_get_request_var( static::$key_current_section_get_var );
+		}
```

Comment:
For this, we need to check specifically for `$_POST[ $current_section_key ]`. We don't want to filter if `$_GET[ $current_section_key ]` exists. We only want to filter on the `POST` request.